json/src/test/kotlin/ktx/json/serializerTest.kt

```diff
@@ -0,0 +1,69 @@
+package ktx.json
+
+import com.badlogic.gdx.utils.Json
+import com.badlogic.gdx.utils.JsonValue
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import io.kotlintest.matchers.shouldEqual
+import org.junit.Test
+
+@Suppress("unused")
+class `should implement KtxReadOnlySerializer with no 'write' method overriden`<T> : KtxReadOnlySerializer<T> {
+  // Guarantees that [write] method is optional to implement.
+
+  override fun read(json: Json, jsonData: JsonValue, type: Class<*>?): T {
+    throw NotImplementedError()
+  }
+}
+
+class ReadOnlySerializerFactoryTest {
+  @Test
+  fun `should return readonly serializer that calls lambda function with 3 parameters`() {
+    val json = Json()
+    val jsonValue = JsonValue(JsonValue.ValueType.`object`)
+    val readFunction: (Json, JsonValue, Class<*>?) -> Data = mock()
+
+    readOnlySerializer(readFunction).read(json, jsonValue, Data::class.java)
+    verify(readFunction).invoke(json, jsonValue, Data::class.java)
+  }
+
+  @Test
+  fun `should return readonly serializer that returns result of calling lambda with 3 parameters`() {
+    val data = Data(intField = 10, stringField = "Test", doubleField = 11.254)
+    val json = Json()
+    val jsonData = JsonValue(JsonValue.ValueType.`object`)
+    val readFunction: (Json, JsonValue, Class<*>?) -> Data = mock {
+      on { invoke(json, jsonData, Data::class.java) }
+          .thenReturn(data)
+    }
```

Comment:
I believe `{ _, _, _ -> data }` lambda would be sufficient, since you don't need to track the invocations in this test. No need to go to far with the mocks.

Reply:
Quite a good idea. Thanks!